crates/cubecl-linalg/src/matmul/kernels/tiling2d/base.rs

```diff
@@ -114,11 +114,8 @@ fn calculate_batch_offsets<N: Numeric>(
 ) -> BatchOffsets {
     let rank = out.rank();
 
-    let dim_m = lhs.shape(rank - 2);
-    let dim_n = rhs.shape(rank - 1);
-
     // Batch offset for output
-    let mut offset_out = dim_m * dim_n * batch_number;
+    let mut offset_out = batch_number * out.stride(rank - 2) * out.shape(rank - 2);
```

Comment:
Can you explain why this still work? Also why you did this change?

Reply:
Because it was broken for non-contiguous out tensors (like ones that are padded for row alignment). This technically only works for out tensors that are contiguous from the second stride onwards, but that's all CUDA does and there's no plan to change this. For tensors that are contiguous from `rank - 2` onwards, `shape(rank - 2) * stride(rank - 2)` equals `stride(rank - 3)`, which is the batch stride. The reason I did it this way is I'm not sure that it's guaranteed that out is at least rank 3, and trying to get the batch stride at `rank - 3` when the tensor is only rank 2 would be an overflow.